WIP Set InsteonEntity name to be combo of description and address.

wonderslug opened the pull request:
## Description:
The Insteon initial entity_id is currently only the unique address of the Insteon device.  This adds the device description to the name to allow easier identification

ie "LampLinc Dimmer 26453a" or "Keypad Dimmer 291abb_2"

**Related issue (if applicable):** fixes #17194 

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable): NA

## Example entry for `configuration.yaml` (if applicable):
```yaml
insteon:
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

wonderslug committed:
Set InsteonEntity name to be combo of description and address.

ie "LampLinc Dimmer 26453a" or "Keypad Dimmer 291abb_2"

wonderslug committed:
Set InsteonEntity name to be combo of description and address.

ie "LampLinc Dimmer 26453a" or "Keypad Dimmer 291abb_2"

wonderslug committed:
Fixed indents

wonderslug committed:
Fixed indents

persandstrom committed:
Verisure standard config for scan interval (#17192)

* verisure configurable polling

* fix indentation

GuyKh committed:
Upgrade boto3 to 1.9.16 (#17140)

cdce8p committed:
Bugfix switch flux - light service call (#17187)

* Bugfix switch flux - light service call

* Change x_val and y_val test

PhilRW committed:
Fix Todoist custom project update (#17115)

* Fix Todoist custom project update

Custom projects were not refreshing the API state and were using
local/stored state. This resulted in invalid tasks being retained upon
update. This change resets the local Todoist API state, syncs it, and
then continues normal update operation(s) on the Todoist project data
object.

* Remove blank line after docstring

* Update logging call

* Simplify logging

persandstrom committed:
vsure version 1.5.0 (#17209)

amelchio committed:
Disallow list/dict for string configuration (#17202)

wonderslug committed:
Enable new registry rename for Insteon (#17171)

* Enable new registry rename for Insteon

* Segment unique_id from name

balloob commented:
This is a breaking change as entity ID was based on name. We should wait with merging this until 84, because after #17171 ships, entity ID will no longer change, reducing the impact of the breaking change.

emontnemery committed:
Fix aliases support for RFLink sensors (#17190)

Danielhiversen committed:
Add forecast for Met.no (#17109)

test met

met no 0.3.0

fix line length

fix line length

Commit:
Add support for ElkM1 alarm/automation panel (#16952)

* Add support for ElkM1 alarm/automation panel.

* fix lines too long

* Address PR comments

* Fix hound ci errors

* Changes for PR comments

* Use vol.Range for checking range value

* Address PR comments

* Fix lint errors

* Added elkm1-lib requirement

* Update coverage to exclude elk

* Fix flake8 errors

* Fix flake8 error

* Cleanup config parsing

* Add housecode converter

* fix PR comments

* fix syntax error

* Fix PR comment

amelchio committed:
Add config entry for LIFX (#17201)

* Add config entry for LIFX

* Use list for dependencies

* Obsolete the platform config

* Use DOMAIN

* Use async_create_task

balloob committed:
Update frontend to 20181007.0

balloob committed:
Update translations

mbrrg committed:
Init sub-components using global var. (#17220)

Danielhiversen committed:
danielhiversen as codeowner for met.no (#17232)

balloob committed:
Prevent accidental device reg override (#17136)

syssi committed:
Add some new model names of Xiaomi Aqara devices (#17234)

* Add additional model name of the Xiaomi Aqara Button (WXKG11LM)

* Add additional model name of the Xiaomi Aqara Wireless Switch (WXKG02LM, WXKG03LM)

* Bump PyXiaomiGateway version

Commit:
Pushed to version 0.7.6 of denonavr library to add more sound modes (#17227)

balloob committed:
Fix SPC (#17236)

syssi committed:
Fix data_key and power_consumed attribute of the Aqara Wall Switch (Closes: #16457) (#17235)

Commit:
Water heater support (#17058)

* Moved econet to water_heater

* Wink and Econet are now in water heater!

* Removed away mode from econet and added demo water heater

* Added demo tests

* Updated coveragerc

* Fix lint issues.

* updated requirements all

* Requirements all actually updated.

* Reset wink and econet and fixed service.

* Reset wink and econet to the correct dev state

* Reset requirements_all and .coveragerc and removed the new econet and wink water_heater files

* Removed @bind_hass service methods

* Actually reset the .coverage file

* Fixed the tests

* Addressed @MartinHjelmare's comments

* Removed unused import

* Switched to async_add_executor_job

* Fixed lint

* Removed is_on

* Added celsius demo water heater and tests.

* Removed metric import

MatteGary committed:
Daikin Climate - Better integration with Climate base component (#16913)

* Daikin Climate - Better integration with Climate base component

Made some modification in order to better integrate the Daikin AC Component with the base Climate Component.
Benefits are:
Support localization for Operation Mode
Support for Homekit Integration (if the AC is turned On, now the status is updated in Homekit)

* Build bug fixing

* Bug fixing in Set Operation_Mode functionality

Fixed to .title() functionality in matching the Operation_Mode

* Fix useless code

* Fix hound bug

* Bug fixing

Change in list of Operation Mode

* Trailing white space fix

* Compile fixing

* Whitespace fix

OttoWinter committed:
Fix potential MQTT discovery race condition (#17208)

* Fix potential MQTT discovery race condition

* Rename data key

dlashua committed:
Add Support for Xiaomi Vibration Sensor (#16422)

OttoWinter committed:
Implement base for MQTT device registry integration (#16943)

* Implement base for MQTT device registry integration

* Lint

* Lint

* Address comments

* Lint

* Lint

* Address comments

* Only add keys if specified

See https://github.com/home-assistant/home-assistant/pull/17136#discussion_r223267185

balloob committed:
Guard for bad device info (#17238)

exxamalte committed:
updated georss-client library to 0.3 (#17239)

schmittx committed:
MyQ cover return unknown state if not available (#17207)

* Add additional supported states

* Use get method for lookup

* Return None if unable to get status

OttoWinter committed:
Add device registry to MQTT binary sensor (#17243)

OttoWinter committed:
Add device registry to MQTT switches (#17244)

OttoWinter committed:
Convert MQTT fan to config entry (#17247)

emontnemery committed:
Add support for HS color to mqtt light (#16958)

* Add support for HS color to mqtt light

* Warn if hs state update is invalid

exxamalte committed:
emptying device state attributes if the update from the feed fails (#17249)

OttoWinter committed:
Add device registry to MQTT cover (#17245)

* Add device registry to MQTT cover

* Fix tests

balloob committed:
Add group foundation (#16935)

Add group foundation

Commit:
Add Elk light platform (#17222)

* Add Elk light platform.

* Add ElkM1 light code. Doh.

* Fix PR comments.

* Fix hound errors.

* Fix PR comment.

* Move state from base to class(s) where used

balloob committed:
Add a webhook automation trigger (#17246)

damarco committed:
Use only_cache parameter in binary_sensor.zha.Remote (#16711)

JariInc committed:
Add configurable temperature step for MQTT climate component (#16201)

* Add configurable temperature step

* Remove temp step from climate component

balloob committed:
block external IP (#17248)

* block external IP

* Update __init__.py

wonderslug committed:
Merge branch 'insteon_initial_entity_name_17194' of github.com:wonderslug/home-assistant into insteon_initial_entity_name_17194

wonderslug committed:
rebased to bring in unqiue_id and centralize network address generation

wonderslug commented:
Rebased to bring in unqiue_id and centralize network address generation to prevent code duplication

wonderslug commented:
Closing in favor of cleaner #17262
